### PR TITLE
Refactoring of and function in ValidationResult class

### PR DESCRIPTION
**Pull request by beingindot**
This function didnt have terminating short circuit 
which is important for and operator functions.

Also this function gets called multiple times since it's validation function. 
hence improving needed here.

After verifying usages, left out following two classes
1. PercentageNumberValidator - since it's using only two validation
2. PhoneNumberValidator - have variable inputs and this is only place.

### CODE REVIEW

**Comment by ripcurlx** [comment]
@beingindot Could you please resolve the conflict? Thanks!

File: core/src/main/java/bisq/core/util/validation/InputValidator.java [hunk 1]
```diff
@@ -21,6 +21,9 @@
 
 import java.math.BigInteger;
 
+import java.util.Arrays;
```
Comment (ripcurlx) [inline core/src/main/java/bisq/core/util/validation/InputValidator.java]: Import statement is not used

**Review by ripcurlx — CHANGES_REQUESTED** [review]
NACK - Please see my comment

**Comment by ripcurlx** [comment]
@beingindot I activated yesterday codacy's PR Quality Review for the Bisq repository. It found for your PR a couple of issues which you can view in the Details link next to the PR check. Please review and push fixes for this issues to trigger a re-review by codacy. Thanks!

**Comment by beingindot** [comment]
@ripcurlx Have removed unused and its passing now. pls review.

**Comment by ripcurlx** [comment]
@beingindot Please make sure that you are applying our format style properly (see my commit https://github.com/bisq-network/bisq/pull/3838/commits/a9bbbdf54566afd623e77ecf348af65e036f16b7). I pushed it this time to save more review time on this, but will reject it in the next PR.

**Review by ripcurlx — APPROVED** [review]
ACK

I tested it on Regtest creating compensation request (BSQ Validator) an offer (all other validators)  and everything works as before.
Still I would appreciate for performance improvements in the future at least comparisons as in https://github.com/bisq-network/bisq/pull/3921#issue-367754081 so we make decisions based on data.

**Pull request merged**